### ISSUE TITLE
Code for DENSE datase

### ISSUE THREAD

**Issue by wangjiyuan9** [md4all/md4all#3]
Thanks for your amazing work in adverse condition MDE!
I found you used DENSE dataset for fog and snow but don't give any code for it .Can you please share it with us to evaluate the models robustness on real snow scenes?

**Comment by morbi25** [comment]
Hello @wangjiyuan9,

Thank you for your interest in our work; we really appreciate that!

Please consider that the experiments we reported for DENSE are only preliminary, thus, in the supplementary material and not in the main paper, as specified in the text. In Section A.3.2 of the paper, we explain why such experiments are preliminary and describe the significant challenges that come with such evaluation due to the limitations of this dataset in the context of depth estimation.
Furthermore, during the execution of these experiments, we encountered major storage limitations due to the size of the dataset. To address this, we opted for a new codebase, which made the DENSE code incompatible with the code for the other datasets shared in this repository. Nevertheless, the hyperparameters were the same. Given the preliminary aspect of these experiments and this code incompatibility, we are not sharing this code at this stage. We may consider adding it later, but we cannot guarantee it due to the amount of work it would require on our side and the limited time available.

If you decide to re-implement those experiments, we would like to clarify a few points below.
- We used the SeeingThroughFog dataset part of DENSE.
- We trained 2 new ForkGAN models, one for day-to-fog and one for day-to-snow. We added images from the Boreas dataset for the snow one, since DENSE did not contain enough.
- We trained a new baseline model on the day-clear samples of the dataset and used it to distill knowledge to our md4all-DD model.
- We created a new torch dataset for SeeingThroughFog, with a similar structure to the torch datasets for nuScenes and RobotCar shared in this repository.
- Since the number of day images for training is relatively small, we increased it by considering the intermediate sweeps. SeeingThroughFog provides for each sample t 9 sweeps. We created more triplets using a sliding window approach. Please keep in mind that this strategy is not ideal, as by doing so, the triplets contain repetitive content with relatively little diversity.
- For the velocity loss, we used the value of "VehSpd_Disp" provided by the CAN data for each sample. Please remember that this value is an approximation across various frames. Therefore, the velocity loss cannot work as reliably as, e.g., for nuScenes.
- As also seen for rain (Figure 2 of the paper), the LiDAR ground truth data often contains artifacts with foggy and snowy weather conditions. We addressed this by filtering the wrong points with a clustering heuristic.
- Other implementation details can be found in Section A.3.2.

If you have any questions, please do not hesitate to contact us again.

Thank you again, and good luck with your research!
